File: app/src/main/java/com/twismart/wallpapershd/ui/base/BaseView.java
package com.twismart.wallpapershd.ui.base;

import android.support.annotation.StringRes;

public interface BaseView {

    void showLoading();

    void hideLoading();

    void onError(@StringRes int resId);

    void onError(String message);

    boolean isNetworkConnected();

    void hideKeyboard();

}

File: app/src/main/kotlin/com/twismart/wallpapershd/ui/base/BaseView.kt
package com.twismart.wallpapershd.ui.base

import android.support.annotation.StringRes

interface BaseView {

    fun showLoading()

    fun hideLoading()

    fun onError(@StringRes resId: Int)

    fun onError(message: String)

    fun isNetworkConnected() : Boolean

    fun hideKeyboard()

}
